### PR TITLE
[Messenger] Deprecate `StopWorkerOnSignalsListener`

### PR DESCRIPTION
**Pull request by HypeMC**
| Q             | A
| ------------- | ---
| Branch?       | 6.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | yes
| Tickets       | -
| License       | MIT
| Doc PR        | -

Followup to #50787.

### CODE REVIEW

**Comment by xabbuh** [comment]
`StopWorkerOnSigtermSignalListener` should be updated as well. It would currently reference the now deprecated class.

**Comment by HypeMC** [comment]
@xabbuh Done. Not sure about the pslam error, it shouldn't be a problem.

File: src/Symfony/Component/Messenger/EventListener/StopWorkerOnSigtermSignalListener.php [hunk 1]
```diff
@@ -11,14 +11,15 @@
 
 namespace Symfony\Component\Messenger\EventListener;
 
-trigger_deprecation('symfony/messenger', '6.3', '"%s" is deprecated, use "%s" instead.', StopWorkerOnSigtermSignalListener::class, StopWorkerOnSignalsListener::class);
+trigger_deprecation('symfony/messenger', '6.3', '"%s" is deprecated, use the "%s" instead.', StopWorkerOnSigtermSignalListener::class, SignalableCommandInterface::class);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/EventListener/StopWorkerOnSigtermSignalListener.php]: this should be moved after the corresponding `use` to work

Comment (HypeMC) [reply src/Symfony/Component/Messenger/EventListener/StopWorkerOnSigtermSignalListener.php]: Fixed, thanks :man_facepalming:

**Comment by nicolas-grekas** [comment]
Thank you @HypeMC.

**Comment by nicolas-grekas** [comment]
PR welcome on 7.0 for the cleanup :pray:

**Comment by xabbuh** [comment]
Looking at what would need to be cleaned up in 7.0 I stumbled upon the `stop_worker_on_signals` messenger config option which until now modifies the `StopWorkerOnSignalsListener` constructor arguments. Shouldn't this part of the `FrameworkExtension` have been updated in this PR too?

**Comment by HypeMC** [comment]
@xabbuh No, the `stop_worker_on_signals` config will be used in certain commands, see #50787.

**Pull request merged**